### PR TITLE
Define "entity", be more deliberate in using entity VS node VS host, and traffic VS flows

### PR DESCRIPTION
**Pull request by theri**
When examining our use of "entity", I found that often we really mean anything that is either a node or measures path properties or processes them and potentially makes decisions based on them, e.g., path selection.
Therefore, I kept "entity" in most places, but added a definition as suggested by Med.

Also chose terms more deliberately in the "Path Selection" section - paths can be selected by any entity, not just the communicating node. And "traffic" is undefined so I substituted it by "flows" when we really just mean flows... or "data" when the packet has not actually been created yet (i.e., before a protocol has been selected).

### CODE REVIEW

File: draft-irtf-panrg-path-properties.md [hunk 1]
```diff
@@ -143,26 +146,26 @@ Therefore, a new technology may implement an existing use case related to differ
 
 ## Path Selection
 
-Entities can choose what traffic to send over which path or subset of paths.
-A node might be able to select between a set of paths, either if there are several paths to the same destination (e.g., in case of a mobile device with two wireless interfaces, both providing a path), or if there are several destinations, and thus several paths, providing the same service (e.g., Application-Layer Traffic Optimization (ALTO) {{RFC5693}}, an application layer peer-to-peer protocol allowing hosts a better-than-random peer selection).
+Nodes may be able to send flows via one out of multiple possible paths, or via a subsets of paths, and an entity may be able to select which path(s) to use.
```
Comment (cyrill-k) [inline draft-irtf-panrg-path-properties.md:149]: Maybe we can rephrase this to clarify that these actions are not exclusive:
"Nodes may be able to send flows via one (or a subset) out of multiple possible paths and an entity may be able to influence the path selection."

File: draft-irtf-panrg-path-properties.md [hunk 2]
```diff
@@ -84,6 +84,9 @@ Note that this document does not assume that any of the listed path properties a
 
 # Terminology
 
+Entity:
+: A physical or virtual device or function which can process packets (i.e., a node or host), measure path properties, or access information about paths. With respect to a given communication (i.e., a flow), an entity may be on the data plane or control plane, and it may be on-path or off-path.
+
 Node:
 : An entity which processes packets, e.g., sends, receives, forwards, or modifies them. A node may be physical or virtual, e.g., a physical device or a service function provided as a virtual element. A node may also be the collection of multiple entities which, as a collection, processes packets, e.g., an entire Autonomous System (AS).
```
[the remark this answers is not in the document]
Comment (cyrill-k) [reply draft-irtf-panrg-path-properties.md:91]: Actually, thinking about this discrepancy a bit more:
Since we added a clear definition of entity, and a node is defined as a collection of entities, it means not every node is an entity.
In the current draft, whenever we mention "entity" in the draft, we want to include any possible node and host.
Now, however, if we talk about an "entity" we only include nodes consisting of a single entity, which might not be what we want in many places of the text.

I think we could solve this by:
- changing the entity and node definition to ensure that every node is an entity (and include in the entity definition that an entity might consist of a collection of physical or virtual devices or functions).
- whenever we talk about an "entity" in the text and want to include the definition of a node, we make sure that we say "a node or an entity". However I think this is not a very elegant solution.

**Review by cyrill-k — CHANGES_REQUESTED** [review]
I added some discussion in the terminology section about the new entity definition.

File: draft-irtf-panrg-path-properties.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (theri) [reply draft-irtf-panrg-path-properties.md:91]: Thanks for catching this.
I think the best way forward is to change the entity and node definition to ensure that every node is an entity.
I was actually intending to make "entities" a superset of "nodes", i.e., an entity is broader than a node, because we need nodes to define a path, and an entity can be basically anything and doesn't have to be on the path.

Changing the "Entity" definition now, and took a stab at simplifying the "Node" definition as well. I don't see the need to define a Node as a collection of entities now, as an entity can already be a collection. Do you agree?

File: draft-irtf-panrg-path-properties.md [hunk 3]
```diff
@@ -84,6 +84,9 @@ Note that this document does not assume that any of the listed path properties a
 
 # Terminology
 
+Entity:
+: A physical or virtual device or function which can process packets (i.e., a node or host), measure path properties, or access information about paths. With respect to a given communication (i.e., a flow), an entity may be on the data plane or control plane, and it may be on-path or off-path.
```
[the remark this answers is not in the document]
Comment (theri) [reply draft-irtf-panrg-path-properties.md:88]: Good idea, will change.

File: draft-irtf-panrg-path-properties.md (hunk 1 above)
Comment (theri) [reply draft-irtf-panrg-path-properties.md:149]: I'm not entirely convinced it said that the actions were exclusive, but I'm fine with shortening it a bit.
I like "influence", but I didn't want to explain path selection with itself, so I went for "influence the decision which path(s) to use". Does this work?

File: draft-irtf-panrg-path-properties.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cyrill-k) [reply draft-irtf-panrg-path-properties.md]: Yes, if we phrase it in this way then it should be clear.

File: draft-irtf-panrg-path-properties.md (hunk 1 above)
Comment (cyrill-k) [reply draft-irtf-panrg-path-properties.md]: Yes, that sounds good

File: draft-irtf-panrg-path-properties.md [hunk 4]
```diff
@@ -85,10 +85,10 @@ Note that this document does not assume that any of the listed path properties a
...
+: A physical or virtual device or function, or a collection of devices or functions, which can, for example, process packets, measure path properties, or access information about paths. With respect to a given communication, an entity may be on the data plane or control plane, and it may be on-path or off-path.
 
 Node:
-: An entity which processes packets, e.g., sends, receives, forwards, or modifies them. A node may be physical or virtual, e.g., a physical device or a service function provided as a virtual element. A node may also be the collection of multiple entities which, as a collection, processes packets, e.g., an entire Autonomous System (AS).
+: An entity which processes packets, e.g., sends, receives, forwards, or modifies them. A node may be physical or virtual, e.g., a physical device or a service function provided as a virtual element. A node may also be an entity which, as a collection, processes packets, e.g., an entire Autonomous System (AS).
```
Comment (cyrill-k) [inline draft-irtf-panrg-path-properties.md:91]: We should maybe clarify what the collection consists of:
"An entity which processes packets, e.g., sends, receives, forwards, or modifies them. A node may be physical or virtual, e.g., a physical device or a service function provided as a virtual element. A node may also be an entity which consists of a collection of devices or functions, e.g., an entire Autonomous System (AS)."

Comment (theri) [reply draft-irtf-panrg-path-properties.md:91]: That works! Then it doesn't say anymore that the collection processes packets, but that's both already part of the definition and clear from the example.

**Comment by theri** [comment]
I think we are aligned now, so I'll merge.
